whatshap/polyphaseplots.py

```diff
@@ -476,3 +479,764 @@ def draw_threading(
+            "Ratio distribution",
+            name1="non-error positions",
+            name2="error positions",
+        )
+        fig = plt.figure(figsize=(12, 9), dpi=100)
+        create_histogram(
+            path[:-4] + ".product-ratio.pdf",
+            product_ratio_non,
+            product_ratio_err,
+            200,
+            [0, 10],
+            "Normalized ratio (progeny and co-parent (product) to parent, 1 = median ratio)",
+            "Ratio distribution",
+            name1="non-error positions",
+            name2="error positions",
+        )
+
+        # Create percentile plots of covered (non-)error positions
+        steps = 100
+        x_lim = 10
+        covered_progeny_non = [0 for _ in range(steps)]
+        covered_progeny_err = [0 for _ in range(steps)]
+        covered_co_parent_non = [0 for _ in range(steps)]
+        covered_co_parent_err = [0 for _ in range(steps)]
+        covered_product_non = [0 for _ in range(steps)]
+        covered_product_err = [0 for _ in range(steps)]
+        for v, s in zip(
+            [
+                covered_progeny_non,
+                covered_progeny_err,
+                covered_co_parent_non,
+                covered_co_parent_err,
+                covered_product_non,
+                covered_product_err,
+            ],
+            [
+                progeny_ratio_non,
+                progeny_ratio_err,
+                co_parent_ratio_non,
+                co_parent_ratio_err,
+                product_ratio_non,
+                product_ratio_err,
+            ],
+        ):
+            last_idx = 0
+            for r in s:
+                idx = int(r * steps / x_lim)
+                if idx >= len(v):
+                    break
+                for i in range(last_idx + 1, idx):
+                    v[i] = v[last_idx]
+                v[idx] = v[last_idx] + 1
+                last_idx = idx
+            for i in range(last_idx + 1, steps):
+                v[i] = v[last_idx]
+
+        fig = plt.figure(figsize=(12, 9), dpi=100)
+        axes = plt.gca()
+        axes.set_xlim([0, x_lim])
+        axes.set_ylim([0, 1])
+        axes.plot(
+            progeny_ratio_non,
+            [(i + 1) / len(progeny_ratio_non) for i in range(len(progeny_ratio_non))],
+            color="tab:cyan",
+        )
+        axes.plot(
+            progeny_ratio_err,
+            [(i + 1) / len(progeny_ratio_err) for i in range(len(progeny_ratio_err))],
+            color="tab:orange",
+        )
+        axes.plot(
+            [x / x_lim for x in range(0, steps)],
+            [
+                b / (a + b) if a + b != 0 else 0
+                for a, b in zip(covered_progeny_non, covered_progeny_err)
+            ],
+            color="tab:green",
+        )
+        plt.title("Ratio percentiles (progeny)")
+        plt.xlabel("Normalized ratio (1 = median ratio)")
+        plt.ylabel("Percentile")
+        savefig(path[:-4] + ".progeny-percentile.pdf", bbox_inches="tight")
```

Comment:
You should be able to use `fig.savefig(...)` here, and then you don’t need the `from pylab import savefig` (pylab is more for interactive use AFAIK).